kubernetes/config/incluster_config.c

```diff
@@ -0,0 +1,154 @@
+#include "incluster_config.h"
+#include <unistd.h>
+#include <fcntl.h>
+#include <errno.h>
+
+#define SERVICE_HOST_ENV_NAME "KUBERNETES_SERVICE_HOST"
+#define SERVICE_PORT_ENV_NAME "KUBERNETES_SERVICE_PORT"
+#define SERVICE_HTTPS_PREFIX "https://"
+#define SERVICE_TOKEN_FILENAME "/var/run/secrets/kubernetes.io/serviceaccount/token"
+#define SERVICE_CERT_FILENAME "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+#define AUTH_TOKEN_KEY "Authorization"
+#define BEARER_TOKEN_TEMPLATE "Bearer %s"
+#define BEARER_TOKEN_BUFFER_SIZE 1024
+
+static int checkServiceAccountFile(const char *fileName)
+{
+    static char fname[] = "checkServiceAccountFile()";
+
+    if (!fileName) {
+        fprintf(stderr, "%s: The file name must be specified.\n", fname);
+        return -1;
+    }
+
+    if (-1 == access(fileName, F_OK)) {
```

Comment:
It seems like this check could be easier if you did:

```c
if (-1 == stat(fileName, &info)) {
  switch (errno) {
     case ENOENT:
         ...
         return -1;
     case EPERM:
         ....
         return-1;
     default:
}
```

Reply:
OK.